go-controller/extensions/pkg/node/node_windows.go

```diff
@@ -24,14 +57,76 @@ func NewNodeHandler(clientset kubernetes.Interface, nodeName string) (types.Node
 // Add function learns about a new node being added to the cluster
 // For a windows node, this means watching for all nodes and programming the routing
 func (n *nodeController) Add(node *kapi.Node) {
+	network, err := hcn.GetNetworkByID(n.networkID)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	nodeSubnet, ok := node.Annotations[extensionHostSubnet]
+
+	if !ok {
+		log.Errorf("Can't retreive the host subnet from the '%s' node's annotations", node.Name)
+		return
+	}
+
+	nodeIPAddress := getFirstInternalIp(node.Status.Addresses)
+
+	if nodeIPAddress == nil {
+		log.Errorf("Can't retreive node '%s' internal IP address from the node's annotations", node.Name)
+		return
+	}
+
+	drmac, ok := node.Annotations[extensionDrMac]
+
+	if !ok {
+		log.Errorf("Can't retreive the distributed router MAC address from the '%s' node's annotations", node.Name)
+		return
+	}
+
+	networkPolicySettings := hcn.RemoteSubnetRoutePolicySetting{
+		IsolationId:                 virtualNetworkIdentifier, // VXLAN virtual network Identifier. Is expected to be 4097 or higher on Windows
+		DistributedRouterMacAddress: drmac,                    // Distributed router/gateway MAC address
+		ProviderAddress:             *nodeIPAddress,           // Host IP address of the node
+		DestinationPrefix:           nodeSubnet,
+	}
+
+	err = AddRemoteSubnetPolicy(network, &networkPolicySettings)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
 	return
 }
 
 func (n *nodeController) Update(oldNode, newNode *kapi.Node) {
+	// TODO: Is this the right approach? In which case is Update() called?
+	n.Delete(oldNode)
+	n.Add(newNode)
 	return
 }
 
 func (n *nodeController) Delete(node *kapi.Node) {
+	network, err := hcn.GetNetworkByID(n.networkID)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+
+	nodeSubnet, ok := node.Annotations[extensionHostSubnet]
```

Comment:
@JocelynBerrendonner note that this won't take into account the case where the annotation is somehow removed from a node, which still needs cleanup on all the other nodes otherwise you'll have dangling subnet policies there. Are you at all able to use a token/cookie/etc to track the policy, rather than the subnet itself?

Reply:
This is a great point! I am still working on a solution for this.

Reply:
This is fixed in the latest iteration. I unfortunately have to store the subnet value for the route, so I used a map to keep track of what was configured. Let me know your thoughts!